### ISSUE TITLE
Grid Power increases on relog

### ISSUE THREAD

**Issue by Serolar** [rwtema/extrautilities_old_issues#1244]
Issue: In singleplayer, If you have any GP generating/draining blocks in the world, return to the main menu, and load back into the world, all GP blocks will duplicate their registry/count. so 1 water mill will turn to 2, then 3, upon repeating the steps. 
Speculation: it will count as though they exist in multitudes. 
Solution: removing all GP blocks from the world, then closing and relaunching the client, will fix the issue

Versions: 
Pack: FTB Presents Direwolf20 : 1.3.2 (Manually Updated with no extra mods)
Minecraft: 1.10.2 
ExtraUtils2: 1.1.3d

**Comment by deadion** [comment]
Same issue same version except even when I start a new world I start with an enormous amount of grid power. Most recently was 8k and it keeps going up. Only an issue because it also seems to recognize my rainbow generator from a previous world that is no longer there.

Side note: Switching dimensions does not fix the issue for me.

**Comment by Hallsie** [comment]
I have a similar issue except it's massively erratic.  The worst It ever was is kind of documented here:  http://imgur.com/a/wJNGe

I don't know what to pull or look for.  I removed all the blocks and restarted the game... no change.  Do they need to be "trashed" possibly?  I don't have that many blocks so I'll try that.  

The first time it happened I thought maybe it was because I used the Builders Wand and that somehow that bugged so I deleted all the blocks I used it on :(

It did fix itself last time EU2 updated on the pack but it's back now.

**Comment by Hallsie** [comment]
Is there a log or debug or something that documents what blocks are placed in the world that I may be able to go into and remove the references to?  I don't know how to provide help on this issue but I would like to help it get resolved.

**Comment by deadion** [comment]
As there is no crash, there is no debug log afaik.

**Comment by InjuredWolf** [comment]
Try finding your 'fml-client-latest.log' file? Or your 'latest.log' file. You don't have to have a crash to have a log. Just close your game and then check those files.
EDIT: You probably won't get any errors, but there's always a possibility.

**Comment by Hallsie** [comment]
Ok so for those that are here.  While I haven't found a permanent fix (mostly because I have no idea what I am doing per-say), I was able to "fix" it with a band-aid fix.

https://www.reddit.com/r/allthemods/comments/5qc26i/eu2_gp_busted_trying_to_solve/dczk783/

I had this on reddit too just in case someone in there knew what to do.

**Comment by AnEvilVegetable** [comment]
Can confirm, I'm at 9k GP using 8x Water Mills because of this. Sometimes my game crashes when I attempt to exit and save my world and I am fairly sure this is the cause. I will post the log if it crashes again.

**Comment by Vectrobe** [comment]
<img width="404" alt="screenshot_36" src="https://cloud.githubusercontent.com/assets/3912867/22870000/cb5a3a8c-f1f7-11e6-93b5-48675cf2edd8.png">

yea, this is a massive mod-breaking bug, the power just keeps growing from nothing and all I've had in the world besides the mill and resonator was a pair of fluid nodes with the "mining" upgrade...

**Comment by garath77** [comment]
Since I'm NOT having an issue like this, it seems to me that this issue COULD be caused by a specific mod conflicting with XU2. Could you all post a list of your mods so we can see if there is one in common?

**Comment by Vectrobe** [comment]
XU2 needs a GP tile dump command for this to be debugged, otherwise there's no way to tell if it's phantom tiles or mathematics causing it.

**Comment by AnEvilVegetable** [comment]
I have not experienced a crash since I last posted but the GP issue persists. Every time I reload the world it multiplies the GP I should have by a factor of n, 128, 256, 384 (or simply additively stacks it idk). If I close the client and restart the game it goes back to normal but the behavior continues if I exit and reload the world without first shutting down the game. I'm playing Sky factory 3 with no additional mods added to it. I've updated all mods and tried reverting Immersive Engineering back to .54 because it was conflicting with Draconic Evolution but it did not help.

**Comment by jk1895** [comment]
I have the same issue. After deleting XU2SaveData.dat it goes away for one relog. Then the GP power increases again after every relog.

**Comment by AutismsDad** [comment]
I have the same issue. Does anyone know a good work around?

**Comment by Vectrobe** [comment]
Delete the data/XU2SaveData.dat inside your save folder every time you load a world other than your main/survival one.

**Comment by Insane96** [comment]
There's any planned fix?

**Comment by 21harrisont** [comment]
I also have this issue. I tried the "fix" but it didn't work! I am NOT starting over!

**Comment by 21harrisont** [comment]
I deleted the world, so there's probably a phantom/ghost file somewhere! Since the world is gone but the fake file exists i probably have to reinstall.
FIX THIS!

**Comment by 21harrisont** [comment]
Wait when trying the fix do I have to delete my transfer nodes that don't use GP? Can I keep them in my inventory so I don't have to recraft it? @Serolar

**Comment by Vectrobe** [comment]
@21harrisont you have to delete the XU2SaveData.dat in the world, then restart the client/server from scratch, or delete the file while neither are running to be sure.

The bug occurs because XU2 loads the data into memory every time a world is loaded, but never re-initialises or clears the data before hand, this has often been seen as a "world leak" by previous mods and MC, but may not actually be world class related.

**Comment by coolman1324** [comment]
I went to the chunk shown in the XU2 data file, and my GP went back to 20 (though there should still be no generation there. It went from the ridiculous amount down to 20 though, meaning it at least recognized that much
![2017-08-13_22 47 27](https://user-images.githubusercontent.com/23014562/29256581-72df7cbc-8079-11e7-988b-a22cf60df9e7.png)
I went back for a few, every second or 2, i increased by 20 again and again.
![2017-08-13_22 46 52](https://user-images.githubusercontent.com/23014562/29256624-ba40429e-8079-11e7-9e4d-cf921a24c640.png)

**Comment by Vectrobe** [comment]
In the latest versions on 1.12 at least, there's a management tool, do you have it there in 1.10?
I suspect though that you have an extra source of GP somewhere that you may have forgotten about...

**Comment by coolman1324** [comment]
No, I don't - It was in a chunk that had never even been generated yet, let alone me visiting it and putting a source of GP there.
